lib/fastMerge.js

```diff
@@ -0,0 +1,45 @@
+// Mostly copied from https://medium.com/@lubaka.a/how-to-remove-lodash-performance-improvement-b306669ad0e1
+
+function isMergeableObject(val) {
+    const nonNullObject = val && typeof val === 'object';
+    return nonNullObject
+    && Object.prototype.toString.call(val) !== '[object RegExp]'
+    && Object.prototype.toString.call(val) !== '[object Date]';
+}
+
+function mergeObject(target, source) {
+    const destination = {};
+    if (isMergeableObject(target)) {
+        // eslint-disable-next-line rulesdir/prefer-underscore-method
+        const targetKeys = Object.keys(target);
+        for (let i = 0; i < targetKeys.length; ++i) {
+            const key = targetKeys[i];
+            destination[key] = target[key];
+        }
+    }
+    // eslint-disable-next-line rulesdir/prefer-underscore-method
+    const sourceKeys = Object.keys(source);
+    for (let i = 0; i < sourceKeys.length; ++i) {
+        const key = sourceKeys[i];
+        if (source[key] === undefined) {
+            // noop
```

Comment:
Maybe we could use `continue` instead of noop but that is just preference

Reply:
That was how I initially implemented it but you have rules that forbid `continue` :D

Reply:
:oh-nothing:

Reply:
I think since this lib is focused on being ultra performant, I would disable to `no-continue` rule and use it to make this as fast as possible.

Reply:
I agree with Tim, I would like to see `continue` here

Reply:
+1 to continue